File: smartcare_mobile/lib/models/appointment.dart
class AppointmentModel {
  final String id;
  final String appointmentId;
  final String patientId;
  final String patientName;
  final String doctorId;
  final String doctorName;
  final String branchName;
  final String departmentName;
  final String appointmentDate;
  final String timeSlot;
  final String status;
  final int queueNumber;
  final bool isPriority;
  final String? cancellationReason;

  AppointmentModel({
    required this.id,
    required this.appointmentId,
    required this.patientId,
    required this.patientName,
    required this.doctorId,
    required this.doctorName,
    required this.branchName,
    required this.departmentName,
    required this.appointmentDate,
    required this.timeSlot,
    required this.status,
    required this.queueNumber,
    this.isPriority = false,
    this.cancellationReason,
  });

  factory AppointmentModel.fromJson(Map<String, dynamic> json) {
    return AppointmentModel(
      id: json['id'] ?? '',
      appointmentId: json['appointmentId'] ?? '',
      patientId: json['patientId'] ?? '',
      patientName: json['patientName'] ?? '',
      doctorId: json['doctorId'] ?? '',
      doctorName: json['doctorName'] ?? '',
      branchName: json['branchName'] ?? 'Central Branch',
      departmentName: json['departmentName'] ?? 'General',
      appointmentDate: json['appointmentDate'] ?? '',
      timeSlot: json['timeSlot'] ?? '',
      status: json['status'] ?? 'WAITING',
      queueNumber: json['queueNumber'] ?? 1,
      isPriority: json['priority'] ?? false,
      cancellationReason: json['cancellationReason'],
    );
  }
}

File: smartcare_backend/src/main/java/com/smartcare/model/Appointment.java
package com.smartcare.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "appointments")
public class Appointment {
    @Id
    private String id;
    private String appointmentId; // e.g. "APT-10024"
    private String patientId;
    private String patientName;
    private String doctorId;
    private String doctorName;
    private String branchId;
    private String branchName;
    private String departmentId;
    private String departmentName;
    private String appointmentDate; // YYYY-MM-DD
    private String timeSlot; // "10:20 - 10:40"
    private AppointmentStatus status;
    private int queuePosition; // Position in waitlist
    private int scheduledSlotNumber; // Assigned slot number for confirmed appointments
    @Builder.Default
    private boolean isPriority = false;
    private String priorityReason;
    private String cancellationReason;
    @Builder.Default
    private String paymentStatus = "UNPAID";
    @Builder.Default
    private double amountPaid = 0.0;
    @Builder.Default
    private Instant createdAt = Instant.now();
    @Builder.Default
    private Instant scheduledAt = null;
    @Builder.Default
    private Instant confirmedAt = null;
    @Builder.Default
    private Instant completedAt = null;
}
